.circleci/config.yml

```
@@ -1,70 +1,168 @@
+              name: "Create $GOPATH directory"
+              command: |
+                sudo mkdir $(go env GOPATH) && sudo chown $USER $(go env GOPATH)
+          - run:
+              name: "Remove downloaded Go files"
+              command: |
+                rm go1.13.4.linux-amd64.tar.gz
+          - run: 
+              name: "Check Go version"
+              command: |
+                go version
+
+      install-tools:
+        steps:
+          - run:
+              name: "Install kind"
+              command: |
+                CGO_ENABLED=0 GO111MODULE="on" go get sigs.k8s.io/kind@v0.7.0
+          - run:
+              name: "Install kubectl"
+              command: |
+                curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl
+                chmod +x ./kubectl
+                sudo mv ./kubectl /usr/local/bin/kubectl
+                kubectl version --client
+
+      create-cluster:
+        parameters:
+          kind-config:
+            type: string
+        steps:
+          - run:
+              name: "Create cluster"
+              command: |
+                cd ~/project
+                
+                if [ -e << parameters.kind-config >> ]; then
+                  cp << parameters.kind-config >> kind-config.yaml
+                else
+                  echo "kind: Cluster" >> kind-config.yaml
+                  echo "apiVersion: kind.sigs.k8s.io/v1alpha3" >> kind-config.yaml
+                fi
+
+                kind create cluster --config kind-config.yaml --image quay.io/giantswarm/kind-node:$KUBERNETES_VERSION
+
+      test-setup:
+        parameters:
+          setup-script:
+            type: string
+        steps:
+          - run:
+              name: "Run test setup commands"
+              command: |
+                if [ -e << parameters.setup-script >> ]; then
+                  echo << parameters.setup-script >>
+                  sh << parameters.setup-script >>
+                fi
+
+      go-test:
+        parameters:
+          env:
+            type: string
+          test-dir:
+            type: string
+        steps:
+          - run:
+              name: "Create .env file"
+              command: |
+                echo '<< parameters.env >>' | sed -e 's/^\(.\)/export \1/' > .env
+          - run:
+              # TODO remove this part after switch to Go modules.
+              name: "Move code to $GOPATH"
+              command: |
+                mkdir -p $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME} && mv ~/project $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
+          - run:
+              name: "Run test"
+              command: |
+                cd $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
+                source .env
+                CGO_ENABLED=0 E2E_KUBECONFIG=~/.kube/config go test -tags k8srequired -timeout 20m -v << parameters.test-dir >>
```

I'd like all steps to basically be oneliners. That's how the orb is designed and that makes it easy to debug things (don't care about the `cd` part because it goes away anyway). 

```suggestion
                # TODO remove `cd` line after switch to Go modules.
                cd $GOPATH/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
                source .env && CGO_ENABLED=0 E2E_KUBECONFIG=~/.kube/config go test -tags k8srequired -timeout 20m -v << parameters.test-dir >>
```

---

Cool I'll get that sorted.